code/datums/actions/actions.dm

```diff
@@ -184,93 +184,125 @@
 			INVOKE_ASYNC(src.owner, src.proc_path, arglist(src.proc_args))
 
 /datum/action/bar/icon/build
-	duration = 30
-	var/obj/item/sheet/sheet
-	var/objtype
-	var/cost
-	var/datum/material/mat
-	var/amount
-	var/objname
-	var/callback = null
-	var/obj/item/sheet/sheet2 // in case you need to pull from more than one sheet
-	var/cost2 // same as above
-	var/spot
-	New(var/obj/item/sheet/csheet, var/cobjtype, var/ccost, var/datum/material/cmat, var/camount, var/cicon, var/cicon_state, var/cobjname, var/post_action_callback = null, var/obj/item/sheet/csheet2, var/ccost2, var/spot)
-		..()
-		icon = cicon
-		icon_state = cicon_state
-		sheet = csheet
-		objtype = cobjtype
-		cost = ccost
-		mat = cmat
-		amount = camount
-		objname = cobjname
-		callback = post_action_callback
-		src.spot = spot
-		if (csheet2)
-			sheet2 = csheet2
-		if (ccost2)
-			cost2 = ccost2
+	/// Custom name of the object for messages if needed, otherwise just the object's initial name
+	var/obj_name
+	var/obj_turf
+	var/obj_type
+	/// Amount of the object which will be made when we're done
+	var/obj_amt
+	var/obj/item/sheet/sheet1
+	var/obj/item/sheet/sheet2
+	var/cost1
+	var/cost2
+	/// Are we using a second sheet (e.g. Glass table parts)
+	var/has_sheet2
+	var/datum/material/obj_mat
+	/// Callback once the thing is constructed
+	var/post_callback
+	//New(var/obj/item/sheet/csheet, var/obj/cobjtype, var/ccost, var/datum/material/cmat, var/camount, var/cicon, var/cicon_state, var/cobjname, var/post_action_callback = null, var/obj/item/sheet/csheet2, var/ccost2, var/spot)
+	New(var/obj/otype, var/target, var/oamt, var/btime, var/s1, var/c1, var/s2, var/c2, var/datum/material/omat, var/c_icon = null, var/c_icon_state = null, var/callback = null, var/name = null)
+		..()
+		resumable = FALSE
```

Comment:
None of this should be resumable and frankly I don't know what actions should be resumable. I expect that it works as I believe it does here.